### PR TITLE
feat(helm): Add bloom configurations

### PR DESCRIPTION
**Pull request by QuentinBisson**
**What this PR does / why we need it**:

This PR adds the possibility to configure bloom gateway and compactor easier than when using structuredConfig

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: production/helm/loki/values.yaml [hunk 1]
```diff
@@ -429,6 +440,14 @@ loki:
   # --  Optional index gateway configuration
   index_gateway:
     mode: simple
+  # --  Optional bloom gateway configuration
+  bloom_gateway:
+    enabled: false
+    client:
+      addresses: '{{ include "loki.indexGatewayAddress" . }}'
```
Comment (z0rc) [inline production/helm/loki/values.yaml:447]: According to https://grafana.com/docs/loki/latest/operations/query-acceleration-blooms/#enable-query-acceleration-with-blooms this should be address of bloom-gateway, not index-gateway.

**Review by z0rc — CHANGES_REQUESTED** [review]

**Comment by z0rc** [comment]
Would it make sense to set by default chart value `loki.storage_config.bloom_shipper.working_directory=/var/loki/blooms`? By default it's `/data/blloms`, which doesn't match statefulset volume mounts.

File: production/helm/loki/values.yaml (hunk 1 above)
Comment (QuentinBisson) [reply production/helm/loki/values.yaml:447]: Oh yes you are right, I'll fix this :(

**Comment by QuentinBisson** [comment]
@z0rc to be honest, I tried this working_directory setting on my deployment but the directory stayed empty and the doc is not clear enough about what it does but I'll do it yes

File: production/helm/loki/values.yaml (hunk 1 above)
Comment (QuentinBisson) [reply production/helm/loki/values.yaml]: It's changed :)

**Comment by z0rc** [comment]
> I'm specifically confused because it's not set for the compactor as well

AFAIK, it's because there is `common.path_prefix` set to `/var/loki` (https://github.com/grafana/loki/blob/main/production/helm/loki/values.yaml#L308). But bloom_shipper doesn't use this prefix for some reason.

> the directory stayed empty

AFAIK it's technically a "temporary" dir that may be used when uploading/downloading/compacting bloom blocks to object storage. So on normal operations files there are removed once processing is done.

**Comment by QuentinBisson** [comment]
Alright this is interesting to know. In that case, would it make sense that the compactor workdir then becomes /var/loki/compactor to match the others?

File: production/helm/loki/values.yaml [hunk 2]
```diff
@@ -277,6 +277,17 @@ loki:
...
+    {{- with .Values.loki.bloom_compactor }}
+    bloom_compactor:
+      {{- tpl (. | toYaml) $ | nindent 4 }}
+    {{- end }}
+
+    {{- with .Values.loki.bloom_gateway }}
+    bloom_gateway:
+      {{- tpl (. | toYaml) $ | nindent 4 }}
+    {{- end }}
+
```
Comment (QuentinBisson) [inline production/helm/loki/values.yaml:289]:
```suggestion
```

**Pull request closed without merging**